windows/windows_online_updates_install/windows_online_updates_install.yml

```diff
@@ -0,0 +1,106 @@
+# Copyright 2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Description:
+#   This test case is used for installing Windows online updates in
+# guest OS through Windows Update. If parameter 'windows_updated_build_num'
+# is not set, this test case will try to search available Windows online updates
+# and install without checking updates categories, updated OS build number.
+#
+- name: windows_online_updates_install
+  hosts: localhost
+  gather_facts: no
+  tasks:
+    - name: "Test case block"
+      block:
+        - name: "Initialize Windows updates log file path and install status"
+          ansible.builtin.set_fact:
+            win_updates_log_file: "C:\\win_updates_log.txt"
+            win_online_updates_installed: false
+
+        - name: "Check parameter 'windows_updated_build_num' value"
+          ansible.builtin.assert:
+            that:
+              - windows_updated_build_num is regex("^[0-9]{5}(\.[0-9]+)?$")
+            fail_msg: >-
+              Parameter 'windows_updated_build_num' is not set to a valid value: "{{ windows_updated_build_num }}",
+              please set it to a valid build number, e.g., '22000', '22000.1', '22000.1234'.
+          when: windows_updated_build_num is defined
+
+        - name: "Test setup"
+          include_tasks: ../setup/test_setup.yml
+
+        - name: "Get guest OS version build before installing Windows updates"
+          include_tasks: ../utils/win_get_os_version.yml
+        - name: "Display guest OS version build before installing Windows updates"
+          ansible.builtin.debug: var=win_os_version_build
+        - name: "Set fact of guest OS build number before installing Windows updates"
+          ansible.builtin.set_fact:
+            win_os_build_before_update: "{{ ('.').join(win_os_version_build.split('.')[2:]) }}"
+        - name: "Check expected OS build number"
+          ansible.builtin.assert:
+            that:
+              - win_os_build_before_update is version(windows_updated_build_num, '<=')
+            fail_msg: >-
+              Before installing Windows updates, guest OS build number is '{{ win_os_build_before_update }}',
+              which is newer than the configured expected build number '{{ windows_updated_build_num }}',
+              please check and change parameter value 'windows_updated_build_num' in vars/test.yml.
+          when: windows_updated_build_num is defined
```

Comment:
If `win_os_build_before_update` is newer than `windows_updated_build_num`, shouldn't the case been skipped?

Reply:
Since `windows_updated_build_num` is the build number user specified and user expect for, if it's not newer than the current build, then it's a user error.

Reply:
Got it. Thanks!